cpp/game_main.cpp

```diff
@@ -307,6 +307,50 @@ GameMain::GameMain(Engine *engine)
 		this->use_set_ability = true;
 		this->ability = ability_type::gather;
 	});
+	this->keybind_context.bind(keybinds::action_t::SPAWN_VILLAGER, [this]() {
+		if (this->construct_mode && this->datamanager.producer_count() > 0) {
+			UnitType &type = *this->datamanager.get_type(590);
+			this->placed_units.new_unit(type, this->players[this->engine->current_player - 1], mousepos_tile.to_phys2().to_phys3());
+		}
+	});
+	this->keybind_context.bind(keybinds::action_t::KILL_UNIT, [this]() {
+		selection.kill_unit();
+	});
+
+	// Villager build commands
+	// TODO place this into separate building menus instead of global hotkeys
+	auto bind_building_key = [this](keybinds::action_t action, int building, int military_building) {
+		this->keybind_context.bind(action, [this, building, military_building]() {
+			if (this->selection.contains_villagers()) {
+				this->building_placement = true;
+				if (this->engine->get_keybind_manager().is_keymod_down(KMOD_LCTRL)) {
+					this->editor_current_building = military_building;
+				} else {
+					this->editor_current_building = building;
+				}
+			}
+		});
+	};
+	bind_building_key(keybinds::action_t::BUILDING_1, 598, 609); // House, barracks
+	bind_building_key(keybinds::action_t::BUILDING_2, 574, 558); // Mill, archery range
+	bind_building_key(keybinds::action_t::BUILDING_3, 616, 581); // Mining camp, stable
+	bind_building_key(keybinds::action_t::BUILDING_4, 611, 580); // Lumber camp, siege workshop
```

Comment:
I was about to say the `z` key was already taken, but instead I'll just move the old `z`, `x` and `c` binds to `ctrl+z` etc., as they're just debugging-related hotkeys anyways.

Also, @Jon0 said town center has id 109, but instead I get moss with the debug message `setting unit type 335 TERRN`. So this doesn't seem to work by setting `editor_current_building` to 109.